### PR TITLE
[Merged by Bors] - proposal signed data was used in id

### PR DESCRIPTION
**Pull request by dshulyak**
> {"L":"ERROR","T":"2023-09-08T15:46:18.944Z","N":"9aa91.blockGenerator","M":"failed to process hare output","node_id":"9aa91374168177589adb11c04a9984f33684e3e3cee72b4298c4d5d56d03dcdb","module":"blockGenerator","layer_id":12,"errmsg":"preprocess fetch layer 12 proposals: hint: proposalDB, hash: 0x1d286d884f4a8699182bde631972d1ad649549aa000000000000000000000000, err: validation reject: incorrect hash: proposal want 1d286d884f, got c2986cf657\nhint: proposalDB, hash: 0xab3ef72044915bb64ef3c7eacfdd98f923a72195000000000000000000000000, err: validation reject: incorrect hash: proposal want ab3ef72044, got 9eafb3985e\nhint: proposalDB, hash: 0x5a0cbb754637b44d5df63ca6938be1fad001ac4b000000000000000000000000, err: validation reject: incorrect hash: proposal want 5a0cbb7546, got ab4994459c","name":"blockGenerator"}

### CODE REVIEW

**Comment by dshulyak** [comment]
bors try

File: proposals/handler.go [hunk 1]
```diff
@@ -272,25 +272,26 @@ func (h *Handler) handleProposal(ctx context.Context, expHash types.Hash32, peer
...
 		return fmt.Errorf("failed to verify proposal signature")
 	}
-	p.ActiveSet = set
 	if !h.edVerifier.Verify(signing.BALLOT, p.Ballot.SmesherID, p.Ballot.SignedBytes(), p.Ballot.Signature) {
 		badSigBallot.Inc()
 		return fmt.Errorf("failed to verify ballot signature")
 	}
 
 	// set the proposal ID when received
+	// p.Initialize uses SignedBytes again
```
Comment (poszu) [inline proposals/handler.go]: It took me a while to understand why you moved active set assignment below. How about:
```suggestion
	// set the proposal ID when received
	// It mustn't contain the active set if layer >= AllowEmptyActiveSet
	// (p.Initialize uses SignedBytes again).
```

**Review by poszu — APPROVED** [review]

**Comment by dshulyak** [comment]
bors try

**Review by fasmat — APPROVED** [review]

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**